platform/darwin/src/MGLShape.h

```diff
@@ -15,6 +20,14 @@ NS_ASSUME_NONNULL_BEGIN
 /** The subtitle of the shape annotation. The default value of this property is `nil`. */
 @property (nonatomic, copy, nullable) NSString *subtitle;
 
+#if TARGET_OS_IPHONE
+
+@property (nonatomic) BOOL canShowCallout;
+
+@property (nonatomic, nullable) UIView <MGLCalloutView> *calloutView;
```

Comment:
I’m not sure supporting callout views on shapes should be within the scope of this PR. We currently rely on `mbgl::AnnotationManager` to perform the spatial query that tells us which annotation is near a tap. That query only returns point annotations, not shapes or lines. Once we reimplement `AnnotationManager` atop a GeoJSON layer – #4839 is a start in that direction – we’ll be able to use #352 to query for shapes and lines as well. We’ll still need to figure out what callout positioning should look like for non-point features.